### PR TITLE
DEP PHP Support in CMS5

### PR DESCRIPTION
**Pull request by sabina-talipova**
## Parent issue
- https://github.com/silverstripe/silverstripe-framework/issues/10402

### CODE REVIEW

**Comment by sabina-talipova** [comment]
Requires 
- https://github.com/silverstripe/silverstripe-elemental-bannerblock/pull/95
- https://github.com/silverstripe/silverstripe-elemental-fileblock/pull/33

File: composer.json [hunk 1]
```diff
@@ -5,12 +5,12 @@
...
-        "php": "^7.4 || ^8.0",
-        "silverstripe/recipe-plugin": "^1",
-        "silverstripe/recipe-cms": "4.x-dev",
-        "dnadesign/silverstripe-elemental": "4.x-dev",
-        "silverstripe/elemental-fileblock": "2.x-dev",
-        "silverstripe/elemental-bannerblock": "2.x-dev"
+        "php": "^8.1",
+        "silverstripe/recipe-plugin": "^2",
+        "silverstripe/recipe-cms": "5.x-dev",
+        "dnadesign/silverstripe-elemental": "5.0.0-alpha2",
```
Comment (maxime-rainville) [inline composer.json:11]:
```suggestion
        "dnadesign/silverstripe-elemental": "^5.0.0-alpha2",
```

Comment (maxime-rainville) [inline composer.json]: Most of the dependencies in recipes need to be targeted to a `.x-dev` branch. When COW tag the recipe it rewrites those dependencies.

```suggestion
        "dnadesign/silverstripe-elemental": "5.x-dev",
```

**Review by maxime-rainville — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (sabina-talipova) [reply composer.json]: DONE.
But we have a CI problem 
```
    - Root composer.json requires silverstripe/elemental-bannerblock 3.x-dev -> satisfiable by silverstripe/elemental-bannerblock[3.x-dev].
    - silverstripe/elemental-bannerblock 3.x-dev requires dnadesign/silverstripe-elemental 5.0.0-alpha2 -> found dnadesign/silverstripe-elemental[5.0.0-alpha2] but it conflicts with your root composer.json require (5.x-dev).
```

**Comment by sabina-talipova** [comment]
Requires PR
- https://github.com/silverstripe/silverstripe-elemental/pull/1034

**Review by emteknetnz — APPROVED** [review]
Unit test failure "Call to undefined method GraphQL\Type\Definition\FieldDefinition::create()" has been seen in various other CMS 5 pull-requests and I don't believe it's related to this PR

**Pull request merged**